src/main/java/io/stargate/sgv2/jsonapi/api/v1/metrics/RequestMetricsFilter.java

```diff
@@ -0,0 +1,99 @@
+package io.stargate.sgv2.jsonapi.api.v1.metrics;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandResult;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import java.util.regex.Pattern;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import org.jboss.resteasy.reactive.server.ServerResponseFilter;
+
+/** The filter for counting HTTP requests per command. = */
+@ApplicationScoped
+public class RequestMetricsFilter {
+
+  // split pattern for the user agent, extract only first part of the agent
+  private static final Pattern USER_AGENT_SPLIT = Pattern.compile("[\\s/]");
+
+  /** The {@link MeterRegistry} to report to. */
+  private final MeterRegistry meterRegistry;
+
+  /** The {@link ObjectMapper} to get command name from request. */
+  private final ObjectMapper objectMapper;
+
+  /** The tag for error being true, created only once. */
+  private final Tag errorTrue = Tag.of("error", "true");
+
+  /** Tag that represent the api module. */
+  private final Tag apiTag = Tag.of("module", "jsonapi");
+
+  /** The tag for error being false, created only once. */
+  private final Tag errorFalse = Tag.of("error", "false");
+
+  /** Default constructor. */
+  @Inject
+  public RequestMetricsFilter(MeterRegistry meterRegistry, ObjectMapper objectMapper) {
+    this.meterRegistry = meterRegistry;
+    this.objectMapper = objectMapper;
+  }
+
+  /**
+   * Filter that this bean produces.
+   *
+   * @param requestContext {@link ContainerRequestContext}
+   * @param responseContext {@link ContainerResponseContext}
+   */
+  @ServerResponseFilter
+  public void record(
```

Comment:
Would it make sense to use different name even if `record` is not technically a reserved word in Java?

Reply:
Will change it to recordMetric